security-analysis/security-analysis-api/src/main/java/com/powsybl/security/monitor/StateMonitor.java

```diff
@@ -0,0 +1,53 @@
+/**
+ * Copyright (c) 2016-2017, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.security.monitor;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * @author Etienne Lesot <etienne.lesot at rte-france.com>
+ */
+public class StateMonitor {
+
+    private final ContingencyContext contingencyContext;
+
+    private final List<String> branchIds;
+
+    private final List<String> voltageLevelIds;
+
+    public ContingencyContext getContingencyContext() {
+        return contingencyContext;
+    }
+
+    public List<String> getBranchIds() {
+        return branchIds;
+    }
+
+    public List<String> getVoltageLevelIds() {
+        return voltageLevelIds;
+    }
+
+    public StateMonitor(ContingencyContext contingencyContext, List<String> branchIds, List<String> voltageLevelIds) {
+        this.contingencyContext = Objects.requireNonNull(contingencyContext);
+        this.branchIds = Objects.requireNonNull(branchIds);
+        this.voltageLevelIds = Objects.requireNonNull(voltageLevelIds);
+    }
+
+    public StateMonitor() {
+        this.contingencyContext = null;
+        this.branchIds = Collections.emptyList();
+        this.voltageLevelIds = Collections.emptyList();
```

Comment:
Call the complete constructor : `this(null, Collections.emptyList(), Collections.emptyList());`